### PR TITLE
ember-data: Update various function arguments

### PR DESCRIPTION
**Pull request by tansongyang**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
    + Why all arguments to `buildURL` can be optional:
        * https://github.com/emberjs/data/blob/v2.0.0/packages/ember-data/lib/adapters/build-url-mixin.js#L52
        * https://github.com/emberjs/data/blob/v2.0.0/packages/ember-data/lib/adapters/build-url-mixin.js#L92
    + Why `serializeIntoHash` `options` can be optional:
        * https://github.com/emberjs/data/blob/v2.0.0/packages/ember-data/lib/serializers/rest-serializer.js#L616
        * https://github.com/emberjs/data/blob/v2.0.0/packages/ember-data/lib/serializers/rest-serializer.js#L589
        * https://github.com/emberjs/data/blob/v2.0.0/packages/ember-data/lib/serializers/json-serializer.js#L962
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/ember-data/index.d.ts [hunk 1]
```diff
@@ -118,11 +118,11 @@ declare module "ember-data" {
...
             buildURL(
-                modelName: string,
-                id: string | any[] | {},
-                snapshot: Snapshot | any[],
-                requestType: string,
-                query: {}
+                modelName?: string,
+                id?: string | any[] | {} | null,
+                snapshot?: Snapshot | any[] | null,
+                requestType?: 'findRecord' | 'findAll' | 'query' | 'queryRecord' | 'findMany' | 'findHasMany' | 'findBelongsTo' | 'createRecord' | 'updateRecord' | 'deleteRecord' | string,
```
Comment (chriskrycho) [inline types/ember-data/index.d.ts]: This reduces to just `string`, so the extra types aren't actually useful here – alas, they don't even give you hints in your editor, unfortunately:

string

**Review by chriskrycho — COMMENTED** [review]
There's one bit to simplify here; the change as a whole is 💯

File: types/ember-data/index.d.ts (hunk 1 above)
Comment (tansongyang) [reply types/ember-data/index.d.ts]: Oh no! I had added `| string` as an afterthought and didn't check that autocomplete still worked. Will revert.

**Review by chriskrycho — APPROVED** [review]

**Comment by chriskrycho** [comment]
Make sure you take a look at https://github.com/DefinitelyTyped/DefinitelyTyped/pull/22328 so you don't end up duplicating!

**Pull request merged**